content/doc/book/managing/casc.adoc

```diff
@@ -0,0 +1,287 @@
+---
+title: Jenkins Configuration as Code (JCasC)
+layout: section
+wip: true
+---
+
+The Jenkins Configuration as Code (JCasC) feature defines Jenkins configuration parameters
+in a human-readable YAML file that can be stored under source code.
+This essentially captures the configuration parameters and values that are used when configuring Jenkins from the web UI.
+The configuration can then be modified by editing this file and then applying it.
+
+Configuring and tuning Jenkins and its plugins is a complex process,
+with dozens of parameters to set within the web UI `manage` section.
+Traditionally, experienced Jenkins administrators
+created Apache Groovy `init` scripts
+to automate the configuration for their Jenkins instances.
+This works but requires in-depth understanding of Jenkins APIs
+and the ability to write Groovy scripts.
+Such scripts are powerful and can do almost anything,
+but they also provide few protections against configuration errors.
+
+JCasC provides the convenience and flexibility
+of configuring controllers without using the UI.
+It does not require more understanding of the configuration parameters
+than is required to configure Jenkins through the UI
+and it provides some checks on the values that are provided.
+
+The JCasC configuration file can be checked into an SCM,
+which enables you to determine who made what modifications to the configuration
+and to roll back to a previous configuration if necessary.
+
+The
+link:https://plugins.jenkins.io/configuration-as-code[Configuration as Code] plugin
+must be installed on the Jenkins controller that you will use to build out your JCasC configuration.
+This plugin is not currently included in the suggested plugins list for Jenkins
+but the Docker image in the
+link:https://github.com/jenkinsci/docker[Docker] container
+includes the Configuration as Code plugin.
+It may also be
+link:https://github.com/jenkinsci/docker#preinstalling-plugins[pre-installed]
+on your instance.
+
+== Viewing the JCasC File
+
+When the Configuration as Code plugin is installed,
+you will see *Configuration as Code* in the *System Configuration* section
+of the *Manage Jenkins* page on your dashboard.
+Click on this link,
+then click on *View Configuration*
+to view the YAML file.
+
+This file is an export of the current configuration on this controller.
+It is *not* intended to offer a directly-usable Jenkins YAML configuration;
```

Comment:
this warning isn't worded the best, it's a bit out of date, generally it doesn't fail for any components these days at least not any mainstream ones. (I get that this is copied from the plugin)

There's certain edge cases which mean you may not be able to round trip it but generally the configuration you receive will work.

Reply:
Note that Darin's video about configuring Jenkins with Ansible and JCasC says this, too.

Here, I reworded it to say that it is usually ready to deploy although you probably want to customize it.  And advise that they push the file to the SCM before making modifications so it is preserved for history.